### PR TITLE
Logger context

### PR DESCRIPTION
**Pull request by darkdef**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ❌

### CODE REVIEW

File: src/Driver/Pdo/AbstractPdoCommand.php [hunk 1]
```diff
@@ -255,21 +255,13 @@ protected function internalGetQueryResult(int $queryMode): mixed
...
-     */
-    protected function logQuery(string $rawSql, string $category): void
-    {
-        $this->logger?->log(LogLevel::INFO, $rawSql, [$category]);
-    }
-
     protected function queryInternal(int $queryMode): mixed
     {
         $logCategory = self::class . '::' . $this->getQueryMode($queryMode);
+        $rawSql = $this->getRawSql();
```
Comment (Tigrov) [inline src/Driver/Pdo/AbstractPdoCommand.php:261]: Do not call `CommandInterface::getRawSql()` if no logger or profiler

See https://github.com/yiisoft/db/pull/781

File: docs/en/connection/logger.md [hunk 2]
```diff
@@ -61,3 +61,29 @@ return [
 ```
 
 For other DBMS refer to ["Create connecton"](/docs/en/README.md#create-connection) section.
+
+## Advanced usage of Logger
+
+If you needed re-define logger messages or increase(decrease) level of log:
```
Comment (samdark) [inline docs/en/connection/logger.md:67]:
```suggestion
If you need to redefine logger messages or increase/decrease logging level:

```

File: docs/en/connection/logger.md [hunk 3]
```diff
@@ -61,3 +61,29 @@ return [
...
+1. Create custom logger class
+2. Use context for detect type of message in method "log"
```
Comment (samdark) [inline docs/en/connection/logger.md:69]:
```suggestion
1. Create a custom logger class
2. Use the context to detect type of the message in the "log" method
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -16,6 +16,7 @@
 - Enh #784: Remove unused code in `AbstractSchema::getTableIndexes()` (@vjik)
 - Bug #788: Fix casting integer to string in `AbstractCommand::getRawSql()` (@Tigrov)
 - Enh #789: Remove unnecessary type casting to array in `AbstractDMLQueryBuilder::getTableUniqueColumnNames()` (@Tigrov)
+- Enh #794: Added category in log context (@darkdef)
```
Comment (samdark) [inline CHANGELOG.md:19]:
```suggestion
- Enh #794: Add category to log context (@darkdef)
```

**Review by samdark — APPROVED** [review]

File: src/Driver/Pdo/AbstractPdoCommand.php [hunk 5]
```diff
@@ -255,21 +255,13 @@ protected function internalGetQueryResult(int $queryMode): mixed
...
-    }
-
     protected function queryInternal(int $queryMode): mixed
     {
         $logCategory = self::class . '::' . $this->getQueryMode($queryMode);
 
         if ($this->logger !== null) {
             $rawSql = $this->getRawSql();
-            $this->logQuery($rawSql, $logCategory);
+            $this->logger->log(LogLevel::INFO, $rawSql, [$logCategory, LogTypes::KEY => LogTypes::TYPE_QUERY]);
```
Comment (Tigrov) [inline src/Driver/Pdo/AbstractPdoCommand.php:264]:
```suggestion
            $this->logger->log(LogLevel::INFO, $rawSql = $this->getRawSql(), [$logCategory, LogTypes::KEY => LogTypes::TYPE_QUERY]);
```

And remove `if ($this->logger !== null) {` condition

**Review by terabytesoftw — APPROVED** [review]

File: src/Driver/Pdo/AbstractPdoCommand.php (hunk 5 above)
Comment (darkdef) [reply src/Driver/Pdo/AbstractPdoCommand.php]: changed

**Review by Tigrov — APPROVED** [review]

File: src/Driver/Pdo/LogTypes.php [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Driver\Pdo;
+
+class LogTypes
```
Comment (vjik) [inline src/Driver/Pdo/LogTypes.php:7]:
```suggestion
final class LogTypes
```

File: src/Driver/Pdo/LogTypes.php [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+{
+    public const KEY = 'log-type';
```
Comment (vjik) [inline src/Driver/Pdo/LogTypes.php:9]: `LogTypes` is bad place for this constant. `LogTypes` by name should contain only types. May be move `KEY` to new class `LogContext::TYPE` ?

It allow simplify constants:

- `LogTypes::TYPE_CONNECTION` → `LogTypes::CONNECTION`
- ...

Also suggest rename class to `LogType`. `LogType::Connection` is more pretty.

File: CHANGELOG.md [hunk 8]
```diff
@@ -16,6 +16,7 @@
 - Enh #784: Remove unused code in `AbstractSchema::getTableIndexes()` (@vjik)
 - Bug #788: Fix casting integer to string in `AbstractCommand::getRawSql()` (@Tigrov)
 - Enh #789: Remove unnecessary type casting to array in `AbstractDMLQueryBuilder::getTableUniqueColumnNames()` (@Tigrov)
+- Enh #794: Add category to log context (@darkdef)
```
Comment (vjik) [inline CHANGELOG.md:19]:
```suggestion
- Enh #794: Add message type to log context (@darkdef)
```

Or need rename in code "type" to "category".

**Review by vjik — CHANGES_REQUESTED** [review]

File: src/Driver/Pdo/LogTypes.php (hunk 7 above)
Comment (darkdef) [reply src/Driver/Pdo/LogTypes.php]: About rename class and simplify constants - good idea. New class for one constatns - KEY, it's overhead. I'm simple rename const KEY to LOG_TYPE.

File: src/Driver/Pdo/LogTypes.php (hunk 6 above)
Comment (darkdef) [reply src/Driver/Pdo/LogTypes.php]: +

File: CHANGELOG.md (hunk 8 above)
Comment (darkdef) [reply CHANGELOG.md]: +

**Review by samdark — APPROVED** [review]

**Review by vjik — CHANGES_REQUESTED** [review]
Need revert `AbstractPdoCommand::logQuery()` and schedule it for a major release.

**Review by vjik — APPROVED** [review]

**Pull request merged**